src/Wallabag/CoreBundle/Repository/EntryRepository.php

```diff
@@ -298,8 +298,11 @@ public function findAllByTagId($userId, $tagId)
      */
     public function findByUrlAndUserId($url, $userId)
     {
+        $url = urldecode($url);
+
         $res = $this->createQueryBuilder('e')
-            ->where('e.url = :url')->setParameter('url', urldecode($url))
+            ->where('e.url = :url')->setParameter('url', $url)
```

Comment:
Shouldn't we add an index on url, givenUrl & user_id? Can you test it locally (hopefully on an heavy database)?